$ The canonical format for Linux kernel patches

This format is defined in → Linux/kernel/source/Documentation/SubmittingPatches

The patch is sent to the corresponding → Linux/kernel/source/MAINTAINERS[maintainers] with such a *Subject* format: `[PATCH] subsystem: summarizing phrase`.

The message body looks like
code {
from: John Doe <jd@somewhere.xyz>

A commit message. This message will be added to the permanent changelog
and describes the patch. It is seperated by an empty line from the
preceeding from: line.

The from line is only needed if the person sending the patch is different
from the sender's email address. This person will be credited as the
author of the patch.

The message is followed by one or more "signed off" lines which then
is followed by a line containing three hyphens.

Signed-off-by: Donald Duck <dd@disneyland.com>
---
The optional text after the three hyphens won't go into the changelog.
It can be used for additional comments etc.
It is followed by the actual diff output:

  arch/arm/boot/dts/keystone.dtsi | 8 ++++++++
  1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/keystone.dtsi b/arch/arm/boot/dts/keystone.dtsi
index 3f272826f537..007648971744 100644
--- a/arch/arm/boot/dts/keystone.dtsi
--- b/arch/arm/boot/dts/keystone.dtsi

   etc
   etc
--
2.1.0
code }


The *summarizing phrase* from the *Subject* becomes a globally-unique
identitifier for that patch. It should not be longer than 75 characters.



{ See also

  → Linux/kernel/source/Documentation/email-clients_txt

}
